ui/pages/asset/components/asset-page.tsx

```diff
@@ -221,28 +243,52 @@ const AssetPage = ({
               <Text variant={TextVariant.headingMd} paddingBottom={4}>
                 {t('tokenDetails')}
               </Text>
-              {renderRow(
-                t('contractAddress'),
-                <AddressCopyButton address={address} shorten />,
-              )}
               <Box
                 display={Display.Flex}
                 flexDirection={FlexDirection.Column}
                 gap={2}
               >
-                {asset.decimals !== undefined &&
-                  renderRow(t('tokenDecimal'), <Text>{asset.decimals}</Text>)}
-                {asset.aggregators && asset.aggregators?.length > 0 && (
+                {type === AssetType.token && (
                   <Box>
-                    <Text
-                      color={TextColor.textAlternative}
-                      variant={TextVariant.bodyMdMedium}
+                    {renderRow(
+                      t('contractAddress'),
+                      <AddressCopyButton address={address} shorten />,
+                    )}
+                    <Box
+                      display={Display.Flex}
+                      flexDirection={FlexDirection.Column}
+                      gap={2}
                     >
-                      {t('tokenList')}
-                    </Text>
-                    <Text>{asset.aggregators?.join(', ')}</Text>
+                      {asset.decimals !== undefined &&
+                        renderRow(
+                          t('tokenDecimal'),
+                          <Text>{asset.decimals}</Text>,
+                        )}
+                      {asset.aggregators && asset.aggregators?.length > 0 && (
```

Comment:
Can this be `asset?.aggregators?.length`

Reply:
If we just use that and the `length` is zero, then React would output a `0` instead of nothing.

Reply:
and if we just do `asset.aggregators?.length > 0 && ...` then we get a TypeScript error that `asset.aggregators.length is possibly undefined`.

however we can certainly remove the optional chaining after the `&&`.

Reply:
@darkwing please let me know what you think